java/ql/src/semmle/code/java/security/Mail.qll

```diff
@@ -0,0 +1,72 @@
+/** Provides classes and predicates to reason about email vulnerabilities. */
+
+import java
+import semmle.code.java.frameworks.Mail
+private import semmle.code.java.frameworks.Properties
+
+/**
+ * The insecure way to set Java properties in mail sessions.
+ * 1. Set the `mail.smtp.auth` property to provide the SMTP Transport with a username and password when connecting to the SMTP server or
+ *    set the `mail.smtp.ssl.socketFactory`/`mail.smtp.ssl.socketFactory.class` property to create an SMTP SSL socket.
+ * 2. No `mail.smtp.ssl.checkserveridentity` property is enabled.
+ */
+predicate isInsecureMailPropertyConfig(VarAccess propertiesVarAccess) {
```

Comment:
The interface to the information captured by this predicate ought to be the `Variable` rather than all the accesses to that variable.
```suggestion
predicate isInsecureMailPropertyConfig(Variable properties) {
```

Reply:
In the query we can then replace `isInsecureMailPropertyConfig(ma.getArgument(0))` with `isInsecureMailPropertyConfig(ma.getArgument(0).(VarAccess).getVariable())`